darwin/DarwinMachine.c

```diff
@@ -105,18 +105,22 @@ Machine* Machine_new(UsersTable* usersTable, uid_t userId) {
    openzfs_sysctl_init(&this->zfs);
    openzfs_sysctl_updateArcStats(&this->zfs);
 
+#if !TARGET_OS_IPHONE
    this->GPUService = IOServiceGetMatchingService(kIOMainPortDefault, IOServiceMatching("IOGPU"));
    if (!this->GPUService) {
       CRT_debug("Cannot initialize IOGPU service");
    }
+#endif
 
    return super;
 }
 
 void Machine_delete(Machine* super) {
    DarwinMachine* this = (DarwinMachine*) super;
 
+#if !TARGET_OS_IPHONE
    IOObjectRelease(this->GPUService);
```

Comment:
The original line here is missing a NULL check, which the initialization code above includes.

Reply:
Sure, I can patch that. ~~It also removes the need of using a `#ifndef`, which I like anyway.~~ No it doesn't. I should have known my code.

Reply:
Well, it's fine in your current code, as that field is not used for the iOS 6 SDK, but for normal MacOS X the code may try to release `this->GPUService`, which may or may not have been properly initialized (to non-NULL) in the accompanying initialization function. That's why I wrote "the original line" with emphasis on "original". ;-)

Because AFAIR, all those IOKit functions aren't NULL-safe. In particular the `*Release` functions often cause trouble when being passed NULL pointers.